include/rootston/view.h

```diff
@@ -70,13 +71,27 @@ struct roots_xwayland_surface {
 	struct wl_listener surface_commit;
 };
 
+struct roots_input_panel_surface {
+	struct roots_view *view;
+
+	struct wl_listener destroy;
+
+	struct wl_listener surface_commit;
+};
+
 enum roots_view_type {
 	ROOTS_WL_SHELL_VIEW,
 	ROOTS_XDG_SHELL_V6_VIEW,
 	ROOTS_XDG_SHELL_VIEW,
 #ifdef WLR_HAS_XWAYLAND
 	ROOTS_XWAYLAND_VIEW,
 #endif
+	ROOTS_INPUT_PANEL_VIEW,
```

Comment:
This should probably not be a view, but an entirely different abstraction (like layer shell does)